Src/HALAL/Services/Communication/UART/UART.cpp

```diff
@@ -44,26 +44,72 @@ bool UART::transmit_next_packet(uint8_t id, RawPacket& packet){
     if((uart->huart->ErrorCode & TXBUSYMASK) == 1)
        return false;
 
-    if (HAL_UART_Transmit_DMA(uart->huart, packet.get_data(), packet.get_size()) != HAL_OK){
+    if (HAL_UART_Transmit_DMA(uart->huart, &data, 1) != HAL_OK){
         return false; //TODO: Warning, Error during transmision
     }
 
     return true;
 }
 
-bool UART::receive_next_packet(uint8_t id, RawPacket& packet){
+bool UART::transmit(uint8_t id, uint8_t* data, int16_t size){
     if (not UART::registered_uart.contains(id))
         return false; //TODO: Error handler
 
     UART::Instance* uart = UART::registered_uart[id];
```

Comment:
```suggestion
    UART::UART_HandleTypeDef* handle = UART::registered_uart[id]->huart;
```
Creo que en esta funcion solo usas el handle, por lo que puedes asignarlo directamente y todos los
`uart->huart->variable` quedan `handle->variable`. Ya que estas podrias cambiarle el nombre de la variable huart de instance a handle, a mi me gusta mas personalmente como veas.